### ISSUE TITLE
GithubPages deployment?

### ISSUE THREAD

**Issue by mayrmartin** [donnemartin/system-design-primer#671]
It would be great to have access to this guide in a structured, well-formatted & searchable manner.
I've seen there is an open issue regarding the automated PDF generation in the GitHub Actions but as far as I can see, there the output is neither really well-formatted nor really good searchable.

A possible solution would be to automatically generate a static webpage using [Docosaurus](https://docusaurus.io/ ) from existing .md files inside GitHub Actions & deploy the page via GitHub Pages. To make the whole content searchable there are available plugins such as [https://www.npmjs.com/package/docusaurus-lunr-search](lunr-search) or even Algolia. There is also a possibility to automatically add a PDF download via [mr-pdf](https://github.com/kohheepeace/mr-pdf).

**Comment by rcalixte** [comment]
@mayrmartin Can you share about the PDF not being searchable enough? The PDF generated [here](https://github.com/donnemartin/system-design-primer/issues/4#issuecomment-1001191308) should support full searching. If there is missing functionality, I would like to see if I can tweak the automation to make it more suitable.

**Comment by mayrmartin** [comment]
The PDF you mentioned is indeed searchable but the PR which implements the autogeneration (PR #632) is still open. It would be great if this generation step could be merged in order to have a PDF which is always up-to-date

**Comment by rcalixte** [comment]
Agreed. It's a hefty PR so we're waiting until @donnemartin has time to review it. Thank you for the feedback though! I've been rebasing, regenerating, and re-uploading the PDF when possible until the PR is merged so you can get a fairly updated copy should anything else be merged in the repo before then. Cheers!
